### PR TITLE
ipn/ipnlocal: sanitize prefs in more notify code paths

### PR DESCRIPTION
**Pull request by maisem**
Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -912,7 +912,7 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
 
 	// Now complete the lock-free parts of what we started while locked.
 	if prefsChanged {
-		b.notifyPrefs(prefs.View())
+		b.send(ipn.Notify{Prefs: ptr.To(prefs.View())})
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:915]: why's this one not cleaned?

File: ipn/ipnlocal/local.go [hunk 2]
```diff
@@ -1826,8 +1812,13 @@ func (b *LocalBackend) pollRequestEngineStatus(ctx context.Context) {
 // If no frontend is connected or API watchers are backed up, the notification
 // is dropped without being delivered.
 //
+// If n contains Prefs, those will be sanitized before being delivered.
+//
 // b.mu must not be held.
 func (b *LocalBackend) send(n ipn.Notify) {
+	if n.Prefs != nil {
+		n.Prefs = ptr.To(stripKeysFromPrefs(*n.Prefs))
+	}
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:1821]: oh, because it's done here now

File: ipn/ipnlocal/local.go [hunk 3]
```diff
@@ -1168,8 +1167,8 @@ func (b *LocalBackend) Start(opts ipn.Options) error {
 		b.logf("Start: already running; sending notify")
 		nm := b.netMap
 		state := b.state
-		b.mu.Unlock()
 		p := b.pm.CurrentPrefs()
+		b.mu.Unlock()
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:1171]: unrelated fix, no? send separately?

**Review by bradfitz — APPROVED** [review]

File: ipn/ipnlocal/local.go (hunk 3 above)
Comment (maisem) [reply ipn/ipnlocal/local.go:1171]: added another commit

Comment (bradfitz) [reply ipn/ipnlocal/local.go:1171]: LGTM 🚢
